extension/content/components/recipes/form/RecipeFormHeader.js

```diff
@@ -1,47 +1,57 @@
 import React from "react";
+import PropTypes from "prop-types";
 import { useHistory, useParams } from "react-router-dom";
-import { Alert, Icon, IconButton } from "rsuite";
+import { Alert, Button, Icon, IconButton, Modal } from "rsuite";
 
 import {
   useEnvironmentState,
   useSelectedNormandyEnvironmentAPI,
 } from "devtools/contexts/environment";
 import { useRecipeDetailsState } from "devtools/contexts/recipeDetails";
+import GenericField from "devtools/components/recipes/form/fields/GenericField";
 
 export default function RecipeFormHeader() {
   const { recipeId } = useParams();
   const { selectedKey: environmentKey } = useEnvironmentState();
   const history = useHistory();
   const { data, importInstructions } = useRecipeDetailsState();
   const normandyApi = useSelectedNormandyEnvironmentAPI();
+  const [showCommentModal, setShowCommentModal] = React.useState(false);
 
   const handleSaveClick = () => {
     try {
       if (importInstructions) {
         throw Error("Import Instructions not empty!");
       }
 
-      const { comment: _omitComment, action, ...cleanedData } = data;
-
-      const requestSave = normandyApi.saveRecipe(recipeId, {
-        ...cleanedData,
-        action_id: action.id,
-      });
-
-      requestSave
-        .then((savedRecipe) => {
-          history.push(`/${environmentKey}/recipes/${savedRecipe.id}`);
-          Alert.success("Changes Saved");
-        })
-        .catch((err) => {
-          console.warn(err.message, err.data);
-          Alert.error(`An Error Occurred: ${JSON.stringify(err.data)}`, 5000);
-        });
+      setShowCommentModal(true);
     } catch (err) {
       Alert.error(err.message);
     }
   };
 
+  const saveRecipe = () => {
```

Comment:
we could potentially add `comment` as an argument on this function and then you don't need to manipulate the context/state and the modal would handle the state of that field itself.